File: tests/fixtures/simple.java
// Java test fixture
package com.example;

import java.util.List;
import java.util.ArrayList;

public class Point {
    private double x;
    private double y;
    
    public static final String VERSION = "1.0.0";
    
    public Point(double x, double y) {
        this.x = x;
        this.y = y;
    }
    
    public double getX() {
        return x;
    }
    
    public double distance(Point other) {
        double dx = this.x - other.x;
        double dy = this.y - other.y;
        return Math.sqrt(dx * dx + dy * dy);
    }
}

interface Drawable {
    void draw();
}

class Shape implements Drawable {
    protected String name;
    
    public Shape(String name) {
        this.name = name;
    }
    
    @Override
    public void draw() {
        System.out.println("Drawing " + name);
    }
}

enum Status {
    OK,
    ERROR
}

public class Main {
    private static int counter = 0;
    
    public static void main(String[] args) {
        System.out.println("Hello, Java!");
    }
    
    private static int add(int a, int b) {
        return a + b;
    }
}

File: tests/fixtures/simple.rs
// Rust test fixture
const VERSION: &str = "1.0.0";

pub struct Point {
    x: f64,
    y: f64,
}

impl Point {
    pub fn new(x: f64, y: f64) -> Self {
        Point { x, y }
    }
    
    pub fn distance(&self, other: &Point) -> f64 {
        ((self.x - other.x).powi(2) + (self.y - other.y).powi(2)).sqrt()
    }
}

trait Drawable {
    fn draw(&self);
}

impl Drawable for Point {
    fn draw(&self) {
        println!("Point at ({}, {})", self.x, self.y);
    }
}

enum Status {
    Ok,
    Error(String),
}

fn add(a: i32, b: i32) -> i32 {
    a + b
}

mod utils {
    pub fn format(s: &str) -> String {
        s.trim().to_string()
    }
}

type Result<T> = std::result::Result<T, String>;

static GLOBAL_CONFIG: &str = "default";

macro_rules! debug_print {
    ($($arg:tt)*) => {
        println!("DEBUG: {}", format!($($arg)*));
    };
}

pub fn main() {
    println!("Hello, Rust!");
}